src/util/pointer_offset_size.cpp

```diff
@@ -538,17 +546,18 @@ mp_integer compute_pointer_offset(
       array_type.id()==ID_array,
       "index into array expected, found "+array_type.id_string());
 
-    mp_integer o=compute_pointer_offset(index_expr.array(), ns);
+    auto o = compute_pointer_offset(index_expr.array(), ns);
 
-    if(o!=-1)
+    if(o.has_value())
     {
-      mp_integer sub_size=
-        pointer_offset_size(array_type.subtype(), ns);
+      auto sub_size = pointer_offset_size(array_type.subtype(), ns);
 
       mp_integer i;
 
-      if(sub_size>0 && !to_integer(index_expr.index(), i))
-        return o+i*sub_size;
+      if(
+        sub_size.has_value() && *sub_size > 0 &&
+        !to_integer(index_expr.index(), i))
+        return (*o) + i * (*sub_size);
```

Comment:
consider putting braces around `return` (preceding multi-line conditional) to improve readability